### ISSUE TITLE
Terminal is tiny when extensions create and show them

### ISSUE THREAD

**Issue by SoumadeepChoudhury** [microsoft/vscode#128342]
Issue Type: <b>Bug</b>

I am facing an issue with the terminal window in VS Code.
The first time I open the terminal window or I run a program snippet it works seamlessly fine but on the next Run the terminal window freezes( only showing colon at the end of the terminal window and nothing else).
I am facing this problem after the recent update to 1.58 ...
Please fix his problem as soon as possible.

VS Code version: Code 1.58.0 (2d23c42a936db1c7b3b06f918cde29561cc47cd6, 2021-07-08T06:54:55.083Z)
OS version: Windows_NT x64 10.0.17763
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Pentium(R) CPU G3220 @ 3.00GHz (2 x 2993)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: disabled_off<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|3.87GB (0.54GB free)|
|Process Argv|--crash-reporter-id 60c62695-e077-4632-943b-c0148c3dc4f1|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (11)</summary>

Extension|Author (truncated)|Version
---|---|---
fluent-icons|mig|0.0.12
python|ms-|2021.6.944021595
vscode-pylance|ms-|2021.7.2
jupyter|ms-|2021.8.1013163132
java|red|0.64.1
vscodeintellicode|Vis|1.2.14
vscode-java-debug|vsc|0.34.0
vscode-java-dependency|vsc|0.18.5
vscode-java-pack|vsc|0.17.0
vscode-java-test|vsc|0.30.1
vscode-maven|vsc|0.31.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspyt653:30270858
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30311713
vspre833:30321513
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
vscus158:30321503
pythonvsuse255:30335481
vscorehov:30309549
vscod805:30301674
pythonvspyt200:30331937
vscextlangct:30333562
binariesv615:30325510
vstea450:30335276
bridge0708:30335490

```

</details>

![image](https://user-images.githubusercontent.com/87203832/125118707-b5737800-e10d-11eb-9b6b-fce4cf4422ed.png)

**Comment by meganrogge** [comment]
Please attach a gif of this happening.

**Comment by SoumadeepChoudhury** [comment]
> Please attach a gif of this happening.

**Comment by SoumadeepChoudhury** [comment]
> Please attach a gif of this happening.

Here's the clip.

https://user-images.githubusercontent.com/87203832/125149876-4b80c000-e159-11eb-86a8-b25fc09e3efe.mp4

**Comment by healertrix** [comment]
facing even worse issue mine is showing nothing 😥
![Capture](https://user-images.githubusercontent.com/75298389/125168007-f8425800-e1c0-11eb-99ba-a8fbccfc8b7b.PNG)

**Comment by starryskyegg** [comment]
The very first run is fine.
The second run will be all black (with some texts flashing at the bottom of terminal), and texts can manifest after ctrl++, ctrl+- or switching to other pages and switching to terminal again. If scrolling up, the originally expected texts will show on the top in a weird format (in a row?). So, actually the outputs are printed twice.
Facing this bug after updating to ver.1.58, too

https://user-images.githubusercontent.com/78290099/125169779-4ca80080-e1de-11eb-9bb3-8459ee0c7056.mp4

**Comment by Sam-Varghese** [comment]
Hey folks, Sam Varghese here. I was also facing this issue after VSCode got updated. I even raised a query at the stack overflow regarding this. Here is a link to the question I raised: https://stackoverflow.com/q/68328179/15446699 From there I came to know that an issue regarding this problem has been raised, thanks to SoumadeepChoudhury. I have also recorded my screen in order to show the problem I am facing. Here is it:


https://user-images.githubusercontent.com/79152648/125171976-ac011300-e1d4-11eb-8211-b29f7a777544.mp4

I use windows 10 and updating VSCode made my terminal behave weirdly as shown.

Hope this problem gets sorted out as soon as possible

**Comment by kartikeyporwal** [comment]
I too am experiencing this very similar issue since the time I updated my VS Code to version 1.58.0 on Ubuntu 20.04.2 LTS yesterday, I can't see the terminal content, on top of that, sometimes, I can't kill the terminal using the icon on the terminal's right side. 
I am opening and closing the terminal multiple time using `CTRL+` ` and reloading VS Code as a workaround to deal with these 2 issues.

**Comment by comiluv** [comment]
I am getting this problem as well since upgrading to 1.58.0
I currently rolled back my installation of VSCode to version 1.57.1 which you can do as well by going to this link: https://code.visualstudio.com/updates/v1_57

**Comment by Sam-Varghese** [comment]
We can get rid of this problem by firstly running the `Toggle Panel` command, and then running our programs.

https://user-images.githubusercontent.com/79152648/125191372-aac8f800-e25f-11eb-8747-2e267c6c7c33.mp4

Because I have noticed that this problem occurs only when we don't have our terminal already opened. So opening the terminal via the `Toggle Panel` command could help us eliminate this problem.

**Comment by SoumadeepChoudhury** [comment]
> We can get rid of this problem by firstly running the `Toggle Panel` command, and then running our programs.
> 
>  2.mp4 
> Because I have noticed that this problem occurs only when we don't have our terminal already opened. So opening the terminal via the `Toggle Panel` command could help us eliminate this problem.

Thank You Sam Varghese a lot for getting a great fix about this problem. Yet I hope Microsoft Team fixes this issue properly & universally anon so that we can use VS Code as previously before the recent update to 1.58.

**Comment by Sam-Varghese** [comment]
You're welcome, happy this solution helped 😊

**Comment by healertrix** [comment]
no dude it isnt working for me ,they really broke the terminal in the latest update

**Comment by Sam-Varghese** [comment]
> no dude it isnt working for me ,they really broke the terminal in the latest update

So that means even the `Toggle Panel` command shows you a blank terminal?

**Comment by healertrix** [comment]
unfortunately yes, but how are these ppl closing the issue without even trying to solve our problem

**Comment by Tyriar** [comment]
Adding to recovery to investigate, I think it's related to https://github.com/microsoft/vscode/issues/128255 and https://github.com/microsoft/vscode/issues/128264

**Comment by healertrix** [comment]
yes its still blank

**Comment by korstiaanS** [comment]
Upgrade to 1.58.1 did not help.

**Comment by Tyriar** [comment]
Can repro:

1. Install python extension
2. Open test.py that has contents `print('abc')`
3. Run the file
4. Type exit
5. Run the file, the terminal will run but it will be really tiny:

![image](https://user-images.githubusercontent.com/2193314/125620997-8dd2cec7-c172-47e3-a19d-d43a8d1fc545.png)

**Comment by Tyriar** [comment]
Root cause is the terminal is initialized with 0x0 dimensions

![image](https://user-images.githubusercontent.com/2193314/125621493-4e6478fa-482f-4a06-94f1-d746394bd4d7.png)

**Comment by Tyriar** [comment]
Repro with terminal-sample:

1. Open window without terminal showing
2. Run `Terminal API: Create Terminal`
3. Run `Terminal API: Show`, it works
4. Close it with the bin icon
5. Run `Terminal API: Create Terminal`
6. Run `Terminal API: Show`, the terminal will be 2x1

**Comment by meganrogge** [comment]
the terminal is now less tiny, but the cursor is in the wrong position 🤔 

![recording (32)](https://user-images.githubusercontent.com/29464607/125719689-3394e17c-f221-48a6-a52f-a59269573bae.gif)

**Comment by Tyriar** [comment]
@meganrogge i think this is as expected, when shown it will change from 80x30 to smaller which will end up getting resized by conpty and sometimes it acts up

**Comment by Sam-Varghese** [comment]
Today my VSCode got updated and all issues that I was facing with the terminal got solved.

Thank you very much Meganrogge and Tyriar for solving all issues with the terminal so fast.

**Comment by SoumadeepChoudhury** [comment]
> Today my VSCode got updated and all issues that I was facing with the terminal got solved.
> 
> Thank you very much Meganrogge and Tyriar for solving all issues with the terminal so fast.

I too got an update to 1.58.2 and all the problems regarding the terminal are fixed, Thanks Megan Rogge and Tyriar for the greater help and look out at our issue.

**Comment by GigaMoksh** [comment]
Thank you guys, my issue also got resolved by updating to 1.58.2. Megan and Tyriar you guys are great 💯

**Comment by healertrix** [comment]
mine still hasnt worked 
i am currently using windows terminal pls sort my problem asap
